modules/oe_whitelabel_extra_project/oe_whitelabel_extra_project.module

```diff
@@ -144,5 +160,147 @@ function _oe_whitelabel_extra_project_preprocess_inpage_nav(array &$variables):
+  $element = $variables['elements']['oe_project_dates'][0] ?? [];
+  if ($element['#theme'] ?? NULL === 'time') {
+    // Project lasts a single day.
+    $start_date_element = $element;
+    $end_date_element = $element;
+  }
+  elseif (isset($element['start_date']['#theme'], $element['end_date']['#theme'])) {
+    // Project lasts multiple days.
+    $start_date_element = $element['start_date'];
+    $end_date_element = $element['end_date'];
+  }
+  else {
+    // Empty or incomplete date range. No progress or status can be shown.
+    return;
+  }
+
+  $status_labels = [t('Planned'), t('Ongoing'), t('Closed')];
+
+  // Values for the 'bcl-project-status' component.
+  // Some values contain placeholders that will be updated with javascript.
+  // This makes sure that tests will fail if js does not run.
+  $variables['project_status_args'] = [
+    // Placeholder value.
+    'status' => 'planned',
+    'start_date' => $start_date_element,
+    'start_label' => t('Start'),
+    'end_date' => $end_date_element,
+    'end_label' => t('End'),
+    'label' => t('Status'),
+    // Placeholder value.
+    'badge' => '&ellipsis;',
+    // Placeholder value, identical to 'planned'.
+    'progress' => 0,
+    'attributes' => new Attribute([
+      'data-start-timestamp' => $t_start,
+      'data-end-timestamp' => $t_end,
+      'data-status-labels' => implode('|', $status_labels),
+      // Hide for non-js users, to avoid showing wrong/outdated information.
+      'class' => ['d-none'],
+    ]),
+  ];
+}
+
+/**
+ * Adds variables for the project contributions chart.
+ *
+ * @param array $variables
+ *   Variables from hook_preprocess_node().
+ */
+function _oe_whitelabel_extra_project_preprocess_contributions(array &$variables): void {
+  $legend_items = [];
+  foreach ([
+    'oe_project_budget' => 'bg-gray-400',
+    'oe_project_budget_eu' => 'bg-primary',
+  ] as $field_name => $bg_class) {
+    $field_element = $variables['elements'][$field_name] ?? NULL;
+    if (!isset($field_element[0])) {
+      continue;
+    }
+    $legend_items[] = [
+      'term' => [
+        [
+          'label' => $field_element['#title'],
+          'color' => $bg_class,
+        ],
+      ],
+      'definition' => [
+        [
+          // Render only the field value, without field wrappers.
+          'label' => $field_element[0],
+        ],
+      ],
+    ];
+  }
+
+  if (!$legend_items) {
+    return;
+  }
+
+  $variables['contributions_args'] = [
+    'corporate_contributions' => NULL,
+    'chart' => FALSE,
+    'legend' => [
+      'variant' => 'horizontal',
+      'items' => $legend_items,
+    ],
+  ];
+
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+  $overall_budget = $node->get('oe_project_budget')->value;
+  $eu_budget = $node->get('oe_project_budget_eu')->value;
+
+  if ($overall_budget === NULL || $eu_budget === NULL
+    || $overall_budget <= 0 || $eu_budget < 0
+    || $overall_budget < $eu_budget
```

Comment:
There is a discussion with a proposed format for this, not yet accepted though.
https://www.drupal.org/project/coding_standards/issues/1539712
The direction is to start each new line with the respective operator.
I support this general direction. I like to see lines starting with the operator.

In this case I cheated a bit to have similar conditions on the same line. But I am ok to split these.

In my own projects I sometimes did this to avoid the first line looking different from the other lines, hoping that the "compiler" will optimize it away. But it is unorthodox and I don't suggest doing it:

```php
if (false
  || $condition1
  || $condition2
) {..}
```

Do we have in-house conventions that the operator should be at the end of a line?

Reply:
Uhm not really, thanks for pointing that issue. I usually use mine above because it aligns all the text, allows to comment out lines without breaking, and it's similar to multi-line function invocations.
Let's keep yours, but split each || in a new line.
Also note that my suggestion drops the check for $eu_budget < 0 as we are checking that for overall budget already, so if overall budget is major than 0 and eu budget is major than overall, it implies that eu is also major than 0. Am I wrong?

Reply:
```php
$eu_budget = -10;
$overall_budget = 50;

if ($overall_budget === NULL  # false
    || $eu_budget === NULL  # false
    || $overall_budget <= 0  # false
    || $eu_budget < 0  # true
    || $overall_budget < $eu_budget  # false
  ) {
    // No pie chart can be drawn with these values.
    return;
  }
```
The condition for `$eu_budget < 0` is the only one that is `true`.
Without the condition, the chart is shown. With the condition, the chart is not shown.

(It is another question whether we should prevent this with validation. But I don't like to rely too much on outside constraints for my code.)

Reply:
Yup you are correct indeed.